packages/flutter_tools/test/src/testbed.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/context.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/context_runner.dart';
+
+export 'package:flutter_tools/src/base/context.dart' show Generator;
+
+final Map<Type, Generator> _testbedDefaults = <Type, Generator>{
+  FileSystem: () => MemoryFileSystem(),
+  Logger: () => BufferLogger(),
+};
+
+/// Manages interaction with the tool injection and runner system.
+class Testbed {
+  /// Creates a new [TestBed]
+  ///
+  /// `overrides` provides more overrides in addition to the test defaults.
+  /// `setup` may be provided to apply mocks within the tool managed zone,
+  /// including any specified overrides.
+  Testbed({Future<void> Function() setup, Map<Type, Generator> overrides})
+    : _setup = setup,
+      _overrides = overrides;
+
+
+  final Future<void> Function() _setup;
+  final Map<Type, Generator> _overrides;
+
+  /// Runs `test` within a tool zone.
+  dynamic run(dynamic Function() test) {
```

Comment:
Does it make sense for this to ever return something other than dynamic? If so, you can put a type parameter on the run method and use that instead of dynamic. i.e.:

```dart
T run<T>(T Function() test) {
  ...
  return runInContext<T>(() {
    return context.run<T>(...);
  });
}
```

Reply:
Done